game/state/rules/agenttype.h

```diff
@@ -14,6 +14,15 @@
 namespace OpenApoc
 {
 
+static const int HIRE_COST_SOLDIER = 600;
```

Comment:
Potential problem here is that old saves won't have any values in them. Then I have to hardcode the check in `GameState::initState()` which will look something like this:
```
if (this->agent_salary.empty())
{
	this->agent_salary = {{AgentType::Role::Soldier, 600},
                                           {AgentType::Role::BioChemist, 800},
                                           {AgentType::Role::Physicist, 800},
                                           {AgentType::Role::Engineer, 800}};
}
```

Reply:
TBH, we're at Alpha, whilst it would be **nice** to have those values; they're not going to come from nowhere

For testing and bug-whacking, users should be starting a new game everytime a feature is added to trunk anyway

Once merged, I'm tempted to say we bump to the Alpha 0.3 series and tell all users to start new games just to reinforce that mindset

Thanks for all your effort with this by the way
I'll review the code fully once I've finished with the current sprite stuff I'm testing for our 3D modellers

Reply:
I certainly don't worry about save compatibility right now - especially for values that aren't really useful for mods (as mods are just 'saves' we load on top of the base state, but stuff like this is new info exposed so no current modding stuff could use it anyway)

It doesn't seem out of the question to have a "save version" - maybe increment a number every time we /know/ we've broken something terrible and tell the user, but I fear people will take that as a "If the number hasn't incremented I expect saves to work", which isn't really something I want to put work into right now.

Maybe post-1.0 we can do better trying to keep this stable, but right now I wouldn't put any effort into compatibility.

Reply:
Alright, I'll keep it in mind in the future. Regarding this particular PR I already added defaults to gamestate init. We can remove them in the future.